### PR TITLE
Replace substring

### PR DESCRIPTION
**Pull request by kimlew**
This lets you changes a substring in a filename of a LibreOffice .odt file.

@kcharter Please review. Thanks.

### CODE REVIEW

File: replace_substring_in_filenames.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    echo "Enter the directory path as the 1st parameter." 
+    exit 1
+  fi
+  if [ ! "$2" ]; then
+    echo "Enter the string to replace as the 2nd parameter." 
+    exit 1
+  fi
+  if [ ! "$3" ]; then
+    replacement_string=''
+  fi
```
Comment (kcharter) [inline replace_substring_in_filenames.sh]: You should be able to remove this third `if` block without breaking your script.

If I'm not mistaken the test `[ ! "$3" ]` will pass only if  `"$3"` evaluates to the empty string. This happens when the user gives no third argument at all, or gives an explicit empty string using quotes (`""` or `''`). Either way, the assignment

```
  replacement_string=$3
```

below on line 42 will give `replacement_string` an empty value. In fact, that assignment proves that this `if` block isn't necessary.

**Review by kcharter — COMMENTED** [review]
Looks good, just a couple of lines that aren't needed.

File: replace_substring_in_filenames.sh (hunk 1 above)
Comment (kimlew) [reply replace_substring_in_filenames.sh]: @kevin Please re-review.
- As you pointed out,  I removed the unneeded 3rd command-line argument check & re-tested

- Also, to correct for another issue I found, to keep spaces entered with user prompt responses, I added before the ```read``` commands of the user prompts:
``` IFS='' ```
  - See: https://stackoverflow.com/questions/7314044/use-bash-to-read-line-by-line-and-keep-space

File: replace_substring_in_filenames.sh [hunk 2]
```diff
@@ -17,6 +17,7 @@
 # Check if arguments are provided. If no arguments given, display user prompts 
 # of what to enter.
 if [ $# -eq 0 ]; then
+  IFS=''
   echo "Type the directory path for filenames that need changes: "
   read directory_path
```
Comment (kcharter) [inline replace_substring_in_filenames.sh]: It's not super-important for your script, but when you change a variable like `IFS` that can affect the behaviour of the shell or its built-in commands like `read`, it's a good idea to save its original value and then restore that original value when you don't need the special behaviour. So you'd do something like

```
  OLD_IFS=$IFS
  IFS=''
  echo "Type the directory path for filenames that need changes: "
  read directory_path

  # other lines removed for brevity
  read replacement_string
  IFS=$OLD_IFS
```

**Review by kcharter — COMMENTED** [review]
I have just one little suggestion, but this looks good to me.

File: replace_substring_in_filenames.sh (hunk 2 above)
Comment (kcharter) [reply replace_substring_in_filenames.sh]: Now when you're done with the `read` calls where you need `IFS` to be empty, you restore `IFS` to its original value.

Comment (kimlew) [reply replace_substring_in_filenames.sh]: @kcharter As you suggested, I added this safeguard and re-tested.

File: replace_substring_in_filenames.sh (hunk 1 above)
Comment (kcharter) [reply replace_substring_in_filenames.sh]: Looks good

**Review by kcharter — APPROVED** [review]
Looks good to me!

**Pull request merged**